src/useradd.c

```diff
@@ -849,6 +846,18 @@ static int get_groups (char *list)
 	return 0;
 }
 
+/*
+ * free_groups - free current user_groups list
+ */
+static void free_groups (void)
+{
+	int i = 0;
+	while (NULL != user_groups[i]) {
+		free(user_groups[i]);
+		user_groups[i++] = NULL;
+	}
+}
+
```

Comment:
This looks like a generic function for freeing a (NULL-terminated) list.  We have list interfaces in <lib/list.c>.

Would you mind calling this API free_list() and moving it to lib/list.c?

Also, the loop looks more like a for loop IMO.  How about this?

```c
void
free_list(char **ls)
{
	for (size_t i = 0; ls != NULL; i++) {
		free(ls[i]);
		ls[i] = NULL;
	}
}
```

Reply:
I don't mind. 

Should adding the `free_list()` function and using it here in `useradd.c` be in the same PR or in two different ones?

Reply:
You can do it in the same PR, in a separate commit.

Reply:
Oh, and it can be simplified into this:

```c
void
free_list(char **ls)
{
	for (size_t i = 0; ls != NULL; i++)
		free(ls[i]);
	ls[0] = NULL;
}